pkg/controller/syncset/syncset_controller.go

```diff
@@ -386,16 +387,28 @@ func findSyncSetInstanceForSelectorSyncSet(name string, syncSetInstances []*hive
 }
 
 func (r *ReconcileSyncSet) syncSetInstanceForSyncSet(cd *hivev1.ClusterDeployment, syncSet *hivev1.SyncSet) (*hivev1.SyncSetInstance, error) {
+	ssiLogger := r.logger.WithField("syncsetName", syncSet.Name)
 	cdRef := metav1.NewControllerRef(cd, hivev1.SchemeGroupVersion.WithKind("ClusterDeployment"))
 	hash, err := r.computeHash(syncSet.Spec)
 	if err != nil {
 		return nil, err
 	}
+
+	ssiLogger.Debug("Apply syncset labels to syncsetinstance")
+	newLabels := controllerutils.CloneLabels(syncSet.Labels)
```

Comment:
@staebler I added this because we discussed last week that "child" objects should inherit their parent's labels.

Is this correct or should I remove this?

Reply:
I was wrong about inheriting parents' labels. It is not widespread practice. I gave the example of labels on a deployment getting inherited down through the replicaset, job, and pod. That is not the case. The labels that get passed down are the labels in the pod spec template of the deployment.

I still think it is a good idea to inherit some labels. For example, I like all of the resources related to a specific cluster deployment to have a label with the name of the cluster deployment. However, I can be persuaded that we should not inherit **all** labels.

For what it's worth, I found some functions in the k8s.io/kubernetes repo that may be helpful. For example, there is already a function to do essentially what `controllerutils.AddLabel` does. See https://github.com/kubernetes/kubernetes/blob/b7394102d6ef778017f2ca4046abbaa23b88c290/pkg/util/labels/labels.go#L56.

Reply:
I wonder if in this case, the right labels to clone would be the ones from the clusterdeployment (or both syncset and clusterdeployment). As it is, I don't see how the syncsetinstance would get a label saying that what clusterdeployment it is for.